src/modules/design/diagram-detail/diagram-detail.ts

```diff
@@ -332,4 +404,69 @@ export class DiagramDetail {
       .publish(environment.events.navBar.noValidationError);
     this._diagramIsInvalid = false;
   }
+
+    /**
+   * In the current implementation this method only checks for UserTasks that have
+   * empty or otherwise not allowed FormData in them.
+   *
+   * If that is the case the method will continue by deleting unused/not allowed
+   * FormData to make sure the diagrams XML is further supported by Camunda.
+   *
+   * TODO: Look further into this if this method is not better placed at the FormsSection
+   * in the Property Panel, also split this into two methods and name them right.
+   */
+  private _dropInvalidFormData(): void {
+    const registry: IElementRegistry = this.bpmnio.modeler.get('elementRegistry');
+
+    registry.forEach((element: IShape) => {
+      if (element.type === 'bpmn:UserTask') {
```

Comment:
Extract Business Rule from `if`

```suggestion
      const elementTypeIsUserTask = element.type === 'bpmn:UserTask';
      if (elementTypeIsUserTask) {
```